### PR TITLE
trigger pull requests for vulnerable dependencies

### PR DESCRIPTION
**Pull request by qrkourier**
This PR asks GitHub to send us PRs to upgrade dependencies with known security vulnerabilities for which a patched version is available.

There are two stages to enable this behavior:

1. this PR
1. in repo settings, enable Dependabot security PRs

image

### CODE REVIEW

File: projects/ziti-console-lib/package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "lottie-web": "^5.12.2",
     "ngx-lottie": "^10.0.0",
     "primeng": "^16.0.2",
-    "vanilla-jsoneditor": "^0.18.2",
+    "vanilla-jsoneditor": "^0.23.4",
```
Comment (qrkourier) [inline projects/ziti-console-lib/package.json:25]: changed to match parent project's version of same dep

File: docker-images/ziti-console-assets/Dockerfile [hunk 2]
```diff
@@ -11,12 +11,17 @@ WORKDIR /usr/src/app
 RUN npm install -g @angular/cli@16.0.0-next.0
 
 COPY . .
-RUN npm install --omit=optional
-RUN ng build ziti-console-lib
-RUN ng build ziti-console --base-href ${BASE_HREF} --configuration "production"
+
```
Comment (qrkourier) [inline docker-images/ziti-console-assets/Dockerfile:14]: this doesn't change the image contents, but accelerates builds by mounting a BuildKit cache

File: README.md [hunk 3]
```diff
@@ -64,7 +63,7 @@ To build and run the application from source, you'll also need to make sure you
 
 | Tool        |      Version |
 | :---:       | :---:        |
-| Node.js     |  =18         |
+| Node.js     | >=18         |
```
Comment (qrkourier) [inline README.md:66]: i.e., v20 is :ok_hand: too

File: README.md [hunk 4]
```diff
@@ -159,8 +156,26 @@ From project Root:
 1. Then in a separate window run & watch changes in the main application **app-ziti-console**
 
     ```bash
-    ng build ziti-console-node --watch
+    ng build ziti-console --watch --base-href '/zac/'
```
Comment (qrkourier) [inline README.md:159]: advice for local dev with the SPA instead of deprecated node server

File: projects/ziti-console-lib/package.json (hunk 1 above)
Comment (rgallettonf) [reply projects/ziti-console-lib/package.json:25]: Actually I think we're up to version `^1.1.2` of the vanilla-jsoneditor, via `package.json` in the `ziti-console` root

File: README.md (hunk 4 above)
Comment (rgallettonf) [reply README.md:153]: Hmm I'm not sure we're quite ready to make this the standard. There's other things/conditionds that are required for a Dev to be able to use the ZAC integration that talks to a controller directly (without the node proxy)

For example, the Controller/Edge API needs to use a trusted TLS cert, or the user has to configure their browser to ignore the failed TLS handshake. Also the `/zac` baseHref only works for hosting ZAC via the controller

File: projects/ziti-console-lib/package.json (hunk 1 above)
Comment (qrkourier) [reply projects/ziti-console-lib/package.json]: Good catch. Fixed.

File: README.md (hunk 4 above)
Comment (qrkourier) [reply README.md]: I reverted those premature changes to the local dev instructions.

**Review by rgallettonf — APPROVED** [review]

**Pull request merged**